aiida/manage/tests/pytest_fixtures.py

```diff
@@ -42,16 +69,260 @@ def aiida_caplog(caplog):
+                    aiida_manager.load_profile(current_profile.name, allow_switch=True)
+
+
+@pytest.fixture(scope='session')
+def config_psql_dos(
+    tmp_path_factory: pytest.TempPathFactory,
+    postgres_cluster: dict[str, str],
+) -> t.Callable[[dict[str, t.Any] | None], dict[str, t.Any]]:
+    """Return a profile configuration for the :class:`~aiida.storage.psql_dos.backend.PsqlDosBackend`."""
+
+    def factory(custom_configuration: dict[str, t.Any] | None = None) -> dict[str, t.Any]:
+        """Return a profile configuration for the :class:`~aiida.storage.psql_dos.backend.PsqlDosBackend`.
+
+        :param custom_configuration: Custom configuration to override default profile configuration.
+        :returns: The profile configuration.
+        """
+        configuration = {
+            'storage': {
+                'backend': 'core.psql_dos',
+                'config': {
+                    **postgres_cluster,
+                    'repository_uri': f'file://{tmp_path_factory.mktemp("repository")}',
+                }
+            }
+        }
+        recursive_merge(configuration, custom_configuration or {})
+        return configuration
+
+    return factory
+
+
+def clear_profile():
+    """Clear the currently loaded profile.
+
+    This ensures that the contents of the profile are reset as well as the ``Manager``, which may hold references to
+    data that will be destroyed. The daemon will also be stopped if it was running.
+    """
+    from aiida.engine.daemon.client import get_daemon_client
+
+    daemon_client = get_daemon_client()
+
+    if daemon_client.is_daemon_running:
+        daemon_client.stop_daemon(wait=True)
+
+    manager = get_manager()
+    manager.get_profile_storage()._clear(recreate_user=True)  # pylint: disable=protected-access
+    manager.get_profile_storage()  # reload the storage connection
+    manager.reset_communicator()
+    manager.reset_runner()
+
+
+@pytest.fixture(scope='session')
+def aiida_profile_factory(
+    aiida_instance: Config,
+    aiida_manager: Manager,
+) -> t.Callable[[dict[str, t.Any]], Profile]:
+    """Create a temporary profile, add it to the config of the loaded AiiDA instance and load the profile.
+
+    The default configuration is complete except for the configuration of the storage, which should be provided through
+    the ``custom_configuration`` argument. The storage will be fully reset and initalised, destroying all data that it
+    contains and recreate the default user, making the profile ready for use.
+    """
+
+    def factory(custom_configuration: dict[str, t.Any]) -> Profile:
+        """Create an isolated AiiDA instance with a temporary and fully loaded profile.
+
+        :param custom_configuration: Custom configuration to override default profile configuration.
+        :returns: The constructed profile.
+        """
+        config = aiida_instance
+        configuration = {
+            'default_user_email': 'test@aiida.local',
```

Comment:
Can do. Not sure it matters though, because it people want to override it, they pass it in the `custom_configuration`. They wouldn't have to start messing around with a module constant.